src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseMetaDataChecker.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.blobrouter.util.TimeZones.EUROPE_LONDON_ZONE_ID;
+
+@Component
+public class LeaseMetaDataChecker {
+
+    private static final Logger logger = getLogger(LeaseAcquirer.class);
+
+    public static final String LEASE_EXPIRATION_TIME = "leaseExpirationTime";
+
+    private final int leaseTimeout;
+
+    public LeaseMetaDataChecker(@Value("${storage-blob-lease-timeout-in-minutes}") int leaseTimeout) {
+        this.leaseTimeout = leaseTimeout;
+    }
+
+    public boolean isReadyToUse(BlobClient blobClient, String leaseId) {
+        Map<String, String> blobMetaData = blobClient.getProperties().getMetadata();
+        String leaseExpirationTime = blobMetaData.get(LEASE_EXPIRATION_TIME);
+        var zipFilename = blobClient.getBlobName();
+        var containerName = blobClient.getContainerName();
+
+        logger.info(
+            "Checking if lease acquired on file {} in container {}. Lease Expiration Time: {}",
+            zipFilename,
+            containerName,
+            leaseExpirationTime
+        );
+
+        if (isMetaDataLeaseExpired(leaseExpirationTime)) {
+            blobMetaData.put(
+                LEASE_EXPIRATION_TIME,
+                LocalDateTime.now(EUROPE_LONDON_ZONE_ID)
+                    .plusMinutes(leaseTimeout).toString()
+            );
+            blobClient.setMetadataWithResponse(
+                blobMetaData,
+                new BlobRequestConditions().setLeaseId(leaseId),
+                null, Context.NONE
+            );
+            return true;
+        } else {
+            logger.info(
+                "Lease already acquired on file {} in container {} less than {} seconds ago. ",
```

Comment:
Something is wrong in this log message. It should be either `ago` and timeout value or `until` and timeout expiration value. And currently it says `seconds` but timeout is in `minutes`.

Reply:
fixed